posts/2023-03-31-embrace-kotlin.adoc

```diff
@@ -0,0 +1,273 @@
+- Scala has more advanced features, such as support of typeclasses, macros, more powerful and extensive pattern matching.
+- Java Interoperability: Kotlin prioritizes seamless Java interoperability, allowing developers to use Kotlin code alongside
+Java code without any issues. Scala has Java interoperability as well, but due to its more complex type system and language features, some constructs might require additional effort to work seamlessly with Java code.
+- Ecosystem and Community: Both Kotlin and Scala have active and growing caommunities.
+However, Kotlin's community is rapidly expanding, particularly in the Android development space, while Scala's community is more focused on big data, data processing, and backend services.
+
+== Understanding Kotlin's syntax
+
+While Kotlin's syntax is inspired by Scala, it's a bit more concise but also Java-like.
+You will quickly get used to the new syntax as it is very similar - I'll show some examples.
+
+- Defining functions:
+
+Scala:
+[source,scala]
+----
+def greet(name: String): String = { "Hello, " + name }
+----
+or
+[source,scala]
+----
+def greet(name: String): String = s"Hello, $name!"
+----
+
+Kotlin:
+[source,kotlin]
+----
+fun greet(name: String): String { return "Hello, " + name }
+----
+or
+[source,kotlin]
+----
+fun greet(name: String): String = "Hello, $name!"
+----
+
+- Immutable values:
+
+Scala:
+[source,scala]
+----
+val limit = 100
+----
+
+Kotlin:
+[source,kotlin]
+----
+val limit = 100
+----
+
+- Mutable values:
+
+Scala:
+[source,scala]
+----
+var counter = 0
+----
+
+Kotlin:
+[source,kotlin]
+----
+var counter = 0
+----
+
+- Data classes
+
+One of the most commonly used features in Scala is the case class, which provides an easy way to create immutable data structures
+with automatically generated equals, hashCode, toString, and copy methods. Data classes in Kotlin do exactly the same.
+
+Scala:
+[source,scala]
+----
+case class Person(name: String, age: Int)
+----
+
+Kotlin:
+[source,kotlin]
+----
+data class Person(val name: String, val age: Int)
+----
+
+- Pattern matching:
+
+Scala has powerful pattern matching capabilities, while Kotlin's when expression serves a similar purpose, albeit with some limitations
+
+Scala:
+[source,scala]
+----
+def matchExample(x: Int): String = x match {
+case 1 => "One"
+case 2 => "Two"
+case _ => "Other"
+}
+----
+
+Kotlin:
+[source,kotlin]
+----
+fun matchExample(x: Int): String = when (x) {
+1 -> "One"
+2 -> "Two"
+else -> "Other"
+}
+----
+
+- Extension functions:
+
+Extension Functions: Kotlin's extension functions allow you to extend existing classes without modifying their source code,
+which is similar to Scala's implicit classes. However, Kotlin's approach is more explicit and less prone to unexpected behavior, because
+in Kotlin, extension functions must be imported explicitly if they are defined in a different package than where they are being used.
```

Comment:
I think grammatically the preposition `than` is still ok, but given the context of location, I would choose `from` 
```suggestion
in Kotlin, extension functions must be imported explicitly if they are defined in a different package from where they are being used.